six/modules/c++/cphd/source/CPHDReader.cpp

```diff
@@ -68,6 +68,7 @@ void CPHDReader::initialize(std::shared_ptr<io::SeekableInputStream> inStream,
     }
 
     mMetadata = CPHDXMLControl(logger.get(), false).fromXML(xmlParser.getDocument(), schemaPaths);
+    mMetadata->setVersion(mFileHeader.getVersion());
```

Comment:
I would expect the XML parser to be able to completely construct the CPHD Metadata. Also, since the version and URI reflect the same information, I'd like to be able to keep them in sync without having to remember to do it. For example, with SICD the Metadata just stores the version, then uses `versionToURI` and `getVersionfromURI` to talk to the XML layer.

Reply:
But yeah, this is basically what I need!

Reply:
So to my understanding, SICD is able to keep track of the version and URI together because the URI is always in the same format "urn:SICD:<version>".
So it's easy to get the version from the uri or vice versa.
Because the URI's for CPHD are formatted differently for each version, I couldn't think of a way to handle the version and the URI as one (even though it would make sense to do so).

One hacky solution might be to do some regex parsing to get the version  from the URI, but even that requires some assumptions about the format of the URI.

Reply:
Could you just hardcode the mapping somewhere?

Reply:
Also, for that same reason I couldn't depend on XMLParser to get the version from the XML to update the version in the metadata object, so I have to get it from the header. And CPHDReader seemed like the best place to do that, but yeah I can see how ugly that solution looks lol

Reply:
Haha yeah that works. Like a text file in the conf directory or something?

Reply:
It should be fine to do it in the code itself. This shouldn't ever change; they'll just publish a new version if they want to do something different.